### PR TITLE
Fix panic in felix flushHostIPUpdates

### PR DESCRIPTION
**Pull request by cyclinder**
## Related issues/PRs

fixes https://github.com/projectcalico/calico/issues/9373

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Fix a panic in Felix when accessing a nil address in flushing host addresses, i.e. flushHostIPUpdates function.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/calc/event_sequencer.go [hunk 1]
```diff
@@ -527,9 +527,13 @@ func (buf *EventSequencer) OnHostIPUpdate(hostname string, ip *net.IP) {
 
 func (buf *EventSequencer) flushHostIPUpdates() {
 	for hostname, hostIP := range buf.pendingHostIPUpdates {
+		hostAddr := ""
+		if hostIP != nil {
+			hostAddr = hostIP.String()
```
[the remark this answers is not in the document]
Comment (cyclinder) [reply felix/calc/event_sequencer.go:532]: thanks for your notice, yes, you are right.

[the remark this answers is not in the document]
Comment (cyclinder) [reply felix/calc/event_sequencer.go]: updated.

**Comment by mazdakn** [comment]
@caseydavenport I've not spent time to find the root cause. If we want to do that then we need to spend sometime to try to reproduce the issue. Given that the issue happened in a dual stack cluster, it might be that IPv6 is enabled, but ipv6 address is missing.

**Comment by cyclinder** [comment]
It may be that there is an ipv6 address on the Calico startup, but then it is missing. when the ipv6 address is missing, I think updating it to an empty string is ok.

**Comment by mazdakn** [comment]
@cyclinder it seems the CI execution is not triggered.  Can you please try pushing a dummy commit to trigger it.

**Comment by mazdakn** [comment]
@cyclinder Thanks again for the fix. I merged and cherry picked this fix to v3.29 and v3.28 since we are going to release v3.29.1 patch release on Monday.

**Pull request merged**